pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -105,6 +105,25 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 	}
 	logger.Infof("set user %q bucket max to %d", p.cephUserName, maxBuckets)
 
+	maxObjects := options.ObjectBucketClaim.Spec.AdditionalConfig["maxObjects"]
```

Comment:
Setting the maxObjects and maxSize is idempotent, right? This would run again whenever the operator reconciles.

Reply:
@travisn : Sorry I didn't understand it completely. Does that mean entire `Provision()` will rerun again right when operator reconciles and so will api's [s3svc.CreateBucket()](https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/bucket/provisioner.go#L93) and [cephObject.SetQuotaUserBucketMax()](https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/bucket/provisioner.go#L101) executed again?

Reply:
@travis after creating a obc, then I restarted the operator pod
```console
I0707 10:13:34.785483       7 manager.go:118] objectbucket.io/provisioner-manager "msg"="starting provisioner"  "name"="rook-ceph.ceph.rook.io
/bucket"
I0707 10:13:34.885909       7 controller.go:210]  "msg"="reconciling claim" "key"="default/ceph-delete-bucket" 
I0707 10:13:34.885982       7 helpers.go:80]  "msg"="getting claim for key" "key"="default/ceph-delete-bucket" 
I0707 10:13:34.888497       7 helpers.go:181]  "msg"="getting ObjectBucketClaim's StorageClass" "key"="default/ceph-delete-bucket" 
I0707 10:13:34.892057       7 helpers.go:186]  "msg"="got StorageClass" "key"="default/ceph-delete-bucket" "name"="rook-ceph-delete-bucket"
I0707 10:13:34.892155       7 helpers.go:63]  "msg"="checking OBC for OB name, this indicates provisioning is complete" "key"="default/ceph-de
lete-bucket" "ceph-delete-bucket"=null
I0707 10:13:34.892206       7 helpers.go:65]  "msg"="provisioning already completed" "key"="default/ceph-delete-bucket" "ObjectBucket"="obc-de
fault-ceph-delete-bucket"
I0707 10:13:34.892259       7 controller.go:246]  "msg"="skipping provision" "key"="default/ceph-delete-bucket
```
Apparently the library itself has check whether OB is created or not and hence this code path is executed only once for the creation of OBC IMO. Am I missing something?